control-plane/api-gateway/common/routefilter.go

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package common
+
+// +kubebuilder:skipversion
+
+type ExternalRouteFilter interface {
```

Comment:
Is there a reason that we need this over say leveraging `client.Object` in the places this is used? Doesn't seem like this has anything route-specific included in it (all `client.Object` instances seem like they'd implent this) and the CRD objects shoud already implement `client.Object`.

Reply:
I think I thought we would need more specific things than just get namespace initially. I'll swap over to client.Object